### DISCUSSION TITLE
opentrack 2023.2.0

### DISCUSSION THREAD

**Discussion by sthalik** [opentrack/opentrack#1706] [Announcements]
### Changes

- fix crash on startup with very long environment `PATH` (issue #1661, reported by @wfurney13).
- video/opencv: apply bugfixes for Windows 11 and enable Media Foundation input.
- tracker/pt: improved color keying for overexposed pixels (issue #1665, by @tombrazier).
- tracker/eyeware-beam now has a 32-bit version (by @eyedav).
- tracker/aruco: enable 1080p resolution.
- filter/nm: new 'Natural Movement' filter (issue #1666, by @tombrazier).
- proto/simconnect: fix high CPU usage on FSX (issue #1670, by @tombrazier).
- filter/hamilton, filter/ewma: minor fixes (by @tombrazier, issues #1671 and #1667).
- proto/vjoystick: link to the fork with a working Windows 11 driver version (issue #1700, reported by @ackalker)

**Comment by djesse** [comment]
Hi. For whatever reason I can't start the provided setup.exe from this release on my Win11 PC. Earlier releases don't have this issue. I tested the setup for 2023.1.0 (which I finally installed), v2022.3.2 and v2.3.11.

**Reply by sthalik** [reply]
Probably antivirus-related. You can extract the portable version over an older installation, however.

**Reply by djesse** [reply]
Oh ... thanks. I'll go with the portable version for now.

**Comment by Buzoter123** [comment]
A huge request to add a negative range, or rather a negative interpretation of the incoming data for asymmetric curves. It's hard to explain, but I'll try. For example, we have a two-axis tracker on a gyroscope. It is necessary that when turning the head 180 degrees, the pilot’s head moves away from the back of the seat. Solution: Set the data source for the Z axis to the Yaw axis and adjust the curves. But now, due to the direct interpretation of Yaw, movement in Z occurs either forward or backward depending on the direction of Yaw. As a result, if I look over my left shoulder, I'm moving away along the Z axis, and if I'm looking over my right shoulder, I'm getting closer. It is not right! A solution to the problem could be an additional checkbox for asymmetric curves “interpret positive input data as negative”

**Reply by sthalik** [reply]
Maybe the invert checkboxes should be made to work before mapping, not after?

**Comment by Buzoter123** [comment]
![image](https://github.com/opentrack/opentrack/assets/144796160/5971c966-eb21-4ecc-a31d-1310cf207619)

The thing is that the checkbox should only work for an asymmetrical curve

**Comment by Buzoter123** [comment]
Try adjusting the view along the axes as I wrote: The initial data for Z is Yaw, and at the same time, looking over the shoulder moves the pilot's head away from the back of the seat, regardless of the direction of rotation along Yaw. If possible, please send *.ini

**Comment by Buzoter123** [comment]
One more wish: add a second Y axis. This is necessary in order to make it easier for the virpil to look back. The fact is that when a real pilot looks back, he almost always tilts his head to look behind the armored head. In the current implementation, in order for the virpil to repeat this movement, he needs to tilt his head in a rather complex way - to the side with a turn back, while squinting at the monitor. It is not comfortable. An additional Y axis will allow you to adjust the head deflection along two incoming channels - directly from Y, as well as from Yaw with the selection of the largest value. The result will be the ability to arbitrarily deflect the head along Y when looking forward, and automatically adjustable deflection along Y when looking back.

**Comment by Buzoter123** [comment]
But first of all, make scripts in Python. Then, perhaps, all this will not be needed. :)
